### PR TITLE
fix(core): Auto cast doubles to ints in lua scripts

### PR DESCRIPTION
**Pull request by dranikpg**

### CODE REVIEW

File: src/core/interpreter.cc [hunk 1]
```diff
@@ -84,14 +93,14 @@ void RedisTranslator::OnString(std::string_view str) {
   ArrayPost();
 }
 
-// Doubles are not supported by Redis, however we can support them.
-// Here is the use-case:
-// local foo = redis.call('zscore', 'myzset', 'one')
-// assert(type(foo) == "number")
+// We return doubles as strings likewise to Redis to maintain compatibility.
```
Comment (romange) [inline src/core/interpreter.cc:96]: can we at least use a run-time flag for this?

File: src/server/multi_test.cc [hunk 2]
```diff
@@ -381,6 +381,10 @@ TEST_F(MultiTest, Eval) {
   ASSERT_THAT(resp, ArrLen(3));
   const auto& arr = resp.GetVec();
   EXPECT_THAT(arr, ElementsAre("a", "b", "c"));
+
+  Run({"zadd", "z1", "123", "a"});
+  resp = Run({"eval", "return redis.call('ZSCORE', KEYS[1], 'a') .. '-is-int'", "1", "z1"});
```
Comment (romange) [inline src/server/multi_test.cc:386]: I still need to understand. What did it return before?

Comment (romange) [reply src/server/multi_test.cc:386]: Specifically, maybe you can show the difference with `tonumber` command? I did not understand from your comment above where DF fails

Comment (dranikpg) [reply src/server/multi_test.cc:386]: 123.0-is-int

File: src/core/interpreter.cc [hunk 3]
```diff
@@ -84,13 +84,20 @@ void RedisTranslator::OnString(std::string_view str) {
...
 
-// Doubles are not supported by Redis, however we can support them.
-// Here is the use-case:
-// local foo = redis.call('zscore', 'myzset', 'one')
-// assert(type(foo) == "number")
 void RedisTranslator::OnDouble(double d) {
+  static constexpr double kConvertEps = std::numeric_limits<double>::epsilon();
+
+  double fractpart, intpart;
+  fractpart = modf(d, &intpart);
```
Comment (dranikpg) [inline src/core/interpreter.cc:91]: I'm not sure epsilon is correct if the fractional part is taken from a large number. Does it hold up to max long?

Comment (dranikpg) [reply src/core/interpreter.cc:91]: They scale in the example for comparing the difference,  https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon and in my mind the modf cant improve on accuracy if a large number is already stored skewed

But I checked in the range of int64-max and it seems that modf(max_int) < double_eps 🔦

Comment (romange) [reply src/core/interpreter.cc:91]: lgtm

**Pull request merged**